src/algorithms/clustering/gmm.py

```diff
@@ -126,3 +130,14 @@ def log_likelihood(self):
                 )
             result += np.log(row_result)
         return result
+
+    def update_metrics(self, labels):
```

Comment:
Can it be extracted as an abstract method to Algorithm class (with optional labels field)?